### PR TITLE
[ITSASU-1258] Choose Seq.empty over Option[Seq] for self employment b…

### PR DESCRIPTION
**Pull request by aug24**
…usinesses.

### CODE REVIEW

File: app/views/agent/CheckYourAnswers.scala.html [hunk 1]
```diff
@@ -152,7 +154,7 @@ <h1 class="govuk-heading-l">
     </dl>
 
     </div>
-
+<p>jr</p>
```
Comment (AlexRimmerHMRC) [inline app/views/agent/CheckYourAnswers.scala.html:157]: Oops, you dropped something

Comment (aug24) [reply app/views/agent/CheckYourAnswers.scala.html]: Thank you.  I shall add "jr" to my list of grep checks before committing!

File: app/models/SummaryModel.scala [hunk 2]
```diff
@@ -141,7 +141,7 @@ case class AgentSummary(incomeSource: Option[IncomeSourceModel] = None,
     BusinessSubscriptionDetailsModel(
       nino,
       accountingPeriodVal.getOrElse(throw new Exception("Accounting period not defined for BusinessSubscriptionDetailsModel")),
-      if (useSelfEmployments) selfEmployments.map(_.filter(_.isComplete)) else None,
+      if (useSelfEmployments) selfEmployments.filter(_.isComplete) else Seq.empty,
```
Comment (AlexRimmerHMRC) [inline app/models/SummaryModel.scala:144]:
```suggestion
      if (useSelfEmployments) Some(selfEmployments.filter(_.isComplete)) else None,
```
```suggestion
      if (useSelfEmployments) selfEmployments.filter(_.isComplete) else Seq.empty,
```

Comment (AlexRimmerHMRC) [inline app/models/SummaryModel.scala:144]:
```suggestion
      if (useSelfEmployments) Some(selfEmployments.filter(_.isComplete)) else None,
```
```suggestion
      if (useSelfEmployments) selfEmployments.filter(_.isComplete) else Seq.empty,
```

File: app/models/common/business/BusinessSubscriptionDetailsModel.scala [hunk 3]
```diff
@@ -22,7 +22,7 @@ import play.api.libs.json.Json
 
 case class BusinessSubscriptionDetailsModel(nino: String,
                                             accountingPeriod: AccountingPeriodModel,
-                                            selfEmploymentsData: Option[Seq[SelfEmploymentData]] = None,
+                                            selfEmploymentsData: Seq[SelfEmploymentData] = Seq.empty,
```
Comment (AlexRimmerHMRC) [inline app/models/common/business/BusinessSubscriptionDetailsModel.scala:25]:
```suggestion
                                            selfEmploymentsData: Option[Seq[SelfEmploymentData]] = None,
```

Comment (AlexRimmerHMRC) [inline app/models/common/business/BusinessSubscriptionDetailsModel.scala:25]:
```suggestion
                                            selfEmploymentsData: Option[Seq[SelfEmploymentData]] = None,
```

**Review by AlexRimmerHMRC — APPROVED** [review]

**Pull request merged**